compiler/plugins/target/ROCM/Dialect/ROCM/Transforms/ApplyBuiltinPDLPatterns.cpp

```diff
@@ -203,6 +204,20 @@ populatePDLModuleFromBuiltin(MLIRContext *context, RewritePatternSet &patterns,
 
 namespace {
 
+SmallVector<llvm::StringRef>
+filterUkernelPatternsByTarget(const SmallVector<llvm::StringRef> &names,
+                              StringRef target) {
+  SmallVector<llvm::StringRef> results;
+  std::string pattern =
+      "ukernel_patterns(_.*)*" + target.str() + "(_.*)*\\.mlir";
+  llvm::Regex regex(pattern);
+  for (auto name : names) {
```

Comment:
```suggestion
  for (StringRef name : names) {
```
In general, we try to use explicit types unless clear from the same line (like cast).